docs/appeng/adrs/0006-web-development-framework-preference.md

```diff
@@ -0,0 +1,92 @@
+- `+` Strong templating tooling
+- `+` Possible to gain a portion of the features typically associated with single page applications, with less overhead by pairing the framework with something like [htmx](https://htmx.org/).
+- `+` Uses the python programming language which enjoys broad community support, and is useful across multiple subpractices like web development and data science.
+- `+` Provides batteries included tooling for authentication and authorization, caching, and an administrative console, among others.
+- `+` Uses Python, which is considered easy to learn and is now frequently taught in both college and bootcamp programs.
+
+#### Cons
+
+- `-` Uses Python, which means learning a new programming language for those who haven't been exposed to it yet.
+- `-` Different languages between backend and frontend if Javascript is needed for client interactions.
+- `-` May need to use additional tooling for asynchronous workload use cases (message queues and workers).
+- `-` Python's dynamic typing may lead to more effort spent on unit tests. But this can be mitigated some through [type hinting](https://docs.python.org/3/library/typing.html)
+
+### Spring
+
+#### Pros
+
+- `+` Mature framework that's been around since 2002.
+- `+` Reactive programming model could be useful for a variety of use cases.
+- `+` Spring boot is good for microservices where use cases call for that.
+- `+` Batteries included tooling for things like authentication and authorization and caching, among others.
+- `+` Java tooling is popular in enterprise spaces
+- `+` Uses Java, which is commonly taught in college programs.
+- `+` Java is statically typed which can give safety that would otherwise be gained through additional unit tests.
+
+#### Cons
+
+- `-` Big complex framework with a lot of capabilities, but may be less suited for rapid prototyping use cases.
+- `-` Different languages between backend and frontend if Javascript is needed for client interactions.
+- `-` Potentially too many features, need solid understanding of the framework to know what's most useful. Effort needed to keep team(s) in alignment on practices.
+- `-` Potentially verbose configuration
+- `-` Might have a steep learning curve for some, making it potentially more difficult to onboard junior practitioners.
+
+### Express.js
+
+#### Pros
+
+- `+` Strong performance for event driven workloads
+- `+` Strong community support
+- `+` Uses Javascript, which is commonly taught in bootcamp programs
+- `+` Flexibility in how applications are structured (not opinionated).
+- `+` Mature framework that's been around since 2010.
+- `+` Can use typescript with this framework, allowing us to keep with [ADR 0001](https://playbook.truss.dev/docs/appeng/adrs/use-typescript)
+
+#### Cons
+
+- `-` Not opinionated, this lack of structure can make it more difficult to keep team(s) on the same page around specific practices and approaches.
+- `-` Middleware(s) can grow to become unwiedly
```

Comment:
I think this can be an issue for any framework, not just unique to Express.

Reply:
That's fair, I went ahead and removed that.